yesqa.py

```diff
@@ -49,7 +49,12 @@ def _remove_comments(tokens: Tokens) -> Tokens:
     for i, token in tokenize_rt.reversed_enumerate(tokens):
         if token.name == 'COMMENT':
             if NOQA_RE.search(token.src):
-                _rewrite_noqa_comment(tokens, i, collections.defaultdict(set))
```

Comment:
I would make a separate function for this entirely -- overloading the purpose of `_rewrite_noqa_comment` for this seems not the greatest

also you can utilize the callback functionaly for `re.Pattern.sub` to not need to double-replace match -- this should make it less error prone